ccx_messaging/consumers/consumer.py

```diff
@@ -374,20 +385,30 @@ def deserialize(self, bytes_):
 
         if isinstance(bytes_, (str, bytes, bytearray)):
             try:
-                msg = json.loads(bytes_)
-                if "service" not in msg or "service" in msg and msg["service"] != self.platform_service:
-                    LOG.debug(AnemicConsumer.OTHER_SERVICE_DEBUG_MESSAGE)
-                    return ""
-                LOG.debug(f"Received message for {self.platform_service} service.")
-                return super().deserialize(bytes_)
-
+                return json.loads(bytes_)
             except json.JSONDecodeError as ex:
                 return CCXMessagingError(f"Unable to decode received message: {ex}")
-
-            except jsonschema.ValidationError as ex:
-                return CCXMessagingError(f"Invalid input message JSON schema: {ex}")
         else:
             return CCXMessagingError(
                 f"Unexpected input message type: {bytes_.__class__.__name__}"
             )
 
+    def run(self):
+        """Execute the consumer logic."""
+        signal(SIGALRM, handle_message_processing_timeout)
+        for msg in self.consumer:
+            headers = dict(msg.headers)
+            if not headers:
+                LOG.debug(AnemicConsumer.NO_HEADER_DEBUG_MESSAGE)
+                continue
+            service = headers.get('service')
+            if service:
+                if service != self.platform_service:
```

Comment:
probably can be shortened with `and`, right?

Reply:
I would still need another if for the "no service" case

Reply:
ah, I thought the `else` is for the 2nd `if`, the LOG indentations confused my eyes :D